app/uk/gov/hmrc/openidconnect/userinfo/connectors/AuthConnector.scala

```diff
@@ -16,35 +16,31 @@
 
 package uk.gov.hmrc.openidconnect.userinfo.connectors
 
-import play.api.Logger
 import uk.gov.hmrc.auth.core.retrieve.{Retrievals, ~}
-import uk.gov.hmrc.auth.core.{AuthorisedFunctions, PlayAuthConnector}
+import uk.gov.hmrc.auth.core.{AuthorisedFunctions, Enrolments, PlayAuthConnector}
+import uk.gov.hmrc.http.{HeaderCarrier, NotFoundException}
 import uk.gov.hmrc.openidconnect.userinfo.config.WSHttp
-import uk.gov.hmrc.openidconnect.userinfo.domain.{Authority, DesUserInfo, Enrolment, UserDetails}
-import uk.gov.hmrc.play.auth.microservice.connectors.ConfidenceLevel._
+import uk.gov.hmrc.openidconnect.userinfo.domain.{Authority, DesUserInfo, UserDetails}
 import uk.gov.hmrc.play.config.ServicesConfig
 
 import scala.concurrent.ExecutionContext.Implicits.global
-import scala.concurrent.{ExecutionContext, Future}
-import uk.gov.hmrc.http.{CoreGet, HeaderCarrier, HttpReads, NotFoundException}
+import scala.concurrent.Future
 
 trait AuthConnector extends AuthorisedFunctions {
-  val authBaseUrl: String
-
-  val http: CoreGet
+  def fetchEnrolments()(implicit headerCarrier: HeaderCarrier): Future[Option[Enrolments]] = {
+    authorised().retrieve(Retrievals.allEnrolments) {
+      enrolments => Future.successful(Some(enrolments))
+    }.recover {
+      case e: NotFoundException => None
+    }
+  }
 
-  def fetchEnrolments(auth: Authority)(implicit headerCarrier: HeaderCarrier): Future[Option[Seq[Enrolment]]] = {
-    auth.enrolments map { enrolmentsUri =>
-      http.GET(s"$authBaseUrl$enrolmentsUri") map { response =>
-        response.json.asOpt[Seq[Enrolment]]
-      } recover {
-        case e: NotFoundException => {
-          None
-        }
-      }
-    } getOrElse {
-      Logger.debug("No enrolment uri.")
-      Future.successful(None)
+  def fetchAuthority()(implicit headerCarrier: HeaderCarrier): Future[Option[Authority]] = {
+    authorised().retrieve(Retrievals.credentials and Retrievals.nino) {
+      case credentials ~ nino => Future.successful(Some(Authority(credentials.providerId, nino)))
```

Comment:
Previously NINO was not explicitly required in this method. Are there any tests starting auth service, creating a user without a nino and retrieving user data (eg. name)?
I'm not sure here but Auth might return 401 if you wan't to retrieve something that desn't exist.

Reply:
Not sure if there's a test for it, I can add an acceptance test to be sure.
The nino Retrieval is an Option[String] though so I assume it will just return None if no nino is found as opposed to throwing an error?

Reply:
Actually there is an acceptance test for this scenario: "return correct fields for scope openid:openid:government-gateway for users without a NINO"

Reply:
OK, that's great. It's ok to merge it.